### ISSUE TITLE
ORA-12154 with OCILIB 4.6.2 and ORACLE 18c (18.0.0.0.0)

### ISSUE THREAD

**Issue by fabe2** [vrogier/ocilib#218]
Salut Vincent,
I recently updated our "old" Win32 C++ application and replaced OCILIB 4.3.1 with the current 4.6.2. All works fine when connecting to an ORACLE 11 (11.2.0.4) instance.
However when attempting to connect to an ORACLE 18c SE2 DB using OCI_ConnectionCreate it fails with ORA-12154.
The TNS is correct and can both be tnsping'ed and used for a connection via SQL*plus.
Our workaround is using the old 4.3.1 where the connection works to both ORACLE 11 and 18.
Can you please help? Thanks in advance!
Best regards, Fabian

**Comment by vrogier** [comment]
hi,

ORA-12154 indicates generally a naming resolution issue that is often independent from code.
It is instead related to environment, tnsname.ora and oracle home location.
I don't see any reason for 4.6.2 to not connect to 18c
Can you double check it is not an environment / configuration issue ?

Regards,

Vincent

**Comment by fabe2** [comment]
Hi Vincent,
thanks for the quick reply. As stated above, both tnsping and SQL*plus connection work with the TNS name specified. And the connection works when simply replacing the ociliba.dll 4.6.2 with 4.3.1. So I am quite sure that it is not an environment / configuration issue. But I cannot exclude that possibility though.
Is 4.6.2 looking somewhere else for the tnsnames.ora? Or, how does it know where to look for it?
Thanks, Fabian

**Comment by vrogier** [comment]
what is the version of the client you are using ?
can you make the ocilib demo app to connect your Oracle18c server ?

**Comment by fabe2** [comment]
Good point, Vincent. The client is "still" 11.2.0.4. But I wonder why connecting with OCILIB 4.6.2/client 11.2.0.4/DB 11.2.0.4 works though! It's just the 4.6.2/11.2.0.4/18 combination that doesn't.
I'll let you know the results of the demo app.

**Comment by fabe2** [comment]
Good point, Vincent. The client is actually version 11.2.0.4. It works however with OCILIB 4.6.2 / client 11.2.0.4 / DB 11.2.0.4, but not OCILIB 4.6.2 / client 11.2.0.4 / DB 18c.
I have tested the minimum OCILIB C++ application. While it works with version 4.3.1 it reports ORA-24300 with 4.6.2! So it might be related to issue #213. As you are going to fix that with version 4.6.3 - is there a preview of it, or when are you planning to release it?

**Comment by vrogier** [comment]
Hi,

it is already fixed in develop-v4.6.3 branch.
v4.6.3 can be released anytime (I was waiting for any other issue in the meanwhile)
I can release it this week if needed

Vincent

**Comment by vrogier** [comment]
about ORA-24300, it occurs when you execute a statement, not when you connect.
thus it means that the sample ocilib c++ application using & 11c client, connects nicely to Oracle 18.
ORA-12154 is at connection time. It seems a different issue to me.

**Comment by fabe2** [comment]
You're right! I could narrow it down. The connection actually works, but when executing a statement the ORA-24300 occurs. Sorry for the confusion.
So I assume 4.6.3 will resolve the issue. It's not urgent, but I'm looking forward to the new release, especially the DLL :-)
Thank you very much!

**Comment by vrogier** [comment]
Duplicate of #213

**Comment by fabe2** [comment]
Hi Vincent,
thanks for releasing version 4.6.3. The issue is resolved! Thank you very much.
Regards,
Fabian
